Lib/test/test_email/test_email.py

```diff
@@ -3313,6 +3316,78 @@ def test_parsing_errors(self):
         # Test email.utils.supports_strict_parsing attribute
         self.assertEqual(email.utils.supports_strict_parsing, True)
 
+    def test_parsing_errors_strict_disabled_via_env_var(self):
+        address = 'alice@example.org )Alice('
+        empty = ('', '')
+
+        # Delete cached default value to make the function
+        # reload the environment variable provided below.
+        try:
+            del utils._cached_strict_addr_parsing
+        except AttributeError:
+            pass
+
+        # Strict disabled via env variable, old behavior expected
+        with EnvironmentVarGuard() as environ:
+            environ["PYTHON_EMAIL_DISABLE_STRICT_ADDR_PARSING"] = "1"
+
+            self.assertEqual(utils.getaddresses([address]),
+                             [('', 'alice@example.org'), ('', ''), ('', 'Alice')])
+            self.assertEqual(utils.parseaddr([address]), ('', address))
+
+        # Clear cache again
+        try:
+            del utils._cached_strict_addr_parsing
+        except AttributeError:
+            pass
+
+        # Default strict=True, empty result expected
+        self.assertEqual(utils.getaddresses([address]), [empty])
+        self.assertEqual(utils.parseaddr([address]), empty)
+
+    @contextlib.contextmanager
+    def _email_strict_parsing_conf(self):
+        """Context for the given email strict parsing configured in config file"""
+        old_filename = utils._EMAIL_CONFIG_FILE
```

Comment:
You can use `test.support.swap_attr()` for that.

Reply:
Great idea, implemented in the last fixup commit.